swap/protocol_test.go

```diff
@@ -549,7 +549,10 @@ func TestSwapRPC(t *testing.T) {
 	defer clean()
 
 	// need to have a dummy contract or the call will fail at `GetParams` due to `NewAPI`
-	swap.contract, err = contract.InstanceAt(common.Address{}, swap.backend)
+	depositAmount := int256.Uint256From(9000 * RetrieveRequestPrice)
+
+	// deploy a chequebook
+	err = testDeploy(context.TODO(), swap, depositAmount)
```

Comment:
could you please explain the reason behind these changes in `TestSwapRPC`?

Reply:
For now, I did not change the implementation

Reply:
no, it's better as it is